### PR TITLE
Add capability to concat two `FileContainers`

### LINKED ISSUE #94: Allow to combine Filecontainers
It would be useful to be able to join File containers, optimally also with the option to deduplicate files. 

An example: 
```python
# Looking for climate model data for two scenarios
scenarios = ["ssp585", "ssp119"]
fc_scens = CMIP_FILEFINDER.find_files(
    variable="tas", scenario=scenarios, model=model, resolution="g025", time_res="ann"
)

# only get the historical members that are also in the future scenarios, but only once
unique_scen_members = fc_scens.df.member.unique()

fc_hist = CMIP_FILEFINDER.find_files(
    variable="tas",
    scenario="historical",
    model=model,
    resolution="g025",
    time_res="ann",
    member=unique_scen_members,
)
```

Now I want to combine these. This can currently be done by concatenating the data frames in the FileContainer manually:
```python
fc_all = FileContainer(pandas.concat([fc_hist.df, fc_scens.df]))
```

This could be implemented cleanly, taking care of all the options available in `pandas.concat`.

### PR DESCRIPTION
**Pull request by veni-vidi-vici-dormivi**
Very simple method to concatenate two `FileContainers` together by concatenating there `DataFrames`.
Closes #94

### CODE REVIEW

File: filefinder/_filefinder.py [hunk 1]
```diff
@@ -651,6 +651,37 @@ def search(self, **query):
...
+
+        if not isinstance(other, FileContainer):
+            raise ValueError("Can only concatenate two FileContainers.")
+
+        if self.df.columns is not other.df.columns:
+            raise ValueError("FileContainers must have the same keys.")
+
+        ret = copy.copy(self)
+        ret.df = pd.concat([self.df, other.df])
+        return ret
```
Comment (mathause) [inline filefinder/_filefinder.py:683]:
```suggestion
        df = pd.concat([self.df, other.df])
        return type(self)(df)
```

`type(self)` is `FileContainer`

**Review by mathause — COMMENTED** [review]
Nice! Do we worry about duplicates?

**Comment by veni-vidi-vici-dormivi** [comment]
> Do we worry about duplicates?

Good question, we could add an option `remove_duplicates=True` ?

**Comment by mathause** [comment]
Good idea, with `True` as default

**Comment by mathause** [comment]
or maybe `drop_duplicates` check what pandas does

**Comment by veni-vidi-vici-dormivi** [comment]
In [`pd.concat`](https://pandas.pydata.org/docs/reference/api/pandas.concat.html) they use `verify_integrity` and if there are duplicates, they raise a `ValueError`... not too happy with that tbh.

**Comment by veni-vidi-vici-dormivi** [comment]
But there is a pandas `drop_duplicates` function so that's certainly the better name.

File: filefinder/tests/test_filecontainer.py [hunk 2]
```diff
@@ -140,6 +140,28 @@ def test_filecontainer_search(example_df, example_fc):
...
+
+    with pytest.raises(ValueError, match="FileContainers must have the same keys"):
+        different_keys_fc = FileContainer(example_fc.df.loc[:, ["model", "scen"]])
+        example_fc.concat(different_keys_fc)
+
+    result = example_fc.concat(example_fc, drop_duplicates=False)
+    expected = pd.concat([example_fc.df, example_fc.df])
+
+    pd.testing.assert_frame_equal(result.df, expected)
+    assert result.__len__() == 10
```
Comment (mathause) [inline filefinder/tests/test_filecontainer.py:156]:
```suggestion
    assert len(result) == 10
```

File: filefinder/tests/test_filecontainer.py [hunk 3]
```diff
@@ -140,6 +140,28 @@ def test_filecontainer_search(example_df, example_fc):
...
+
+    result = example_fc.concat(example_fc, drop_duplicates=True)
+    expected = example_fc
+
+    pd.testing.assert_frame_equal(result.df, expected.df)
+    assert result.__len__() == 5
```
Comment (mathause) [inline filefinder/tests/test_filecontainer.py:162]:
```suggestion
    assert len(result) == 5
```

**Review by mathause — APPROVED** [review]
Great!

File: filefinder/tests/test_filecontainer.py (hunk 2 above)
Comment (veni-vidi-vici-dormivi) [reply filefinder/tests/test_filecontainer.py:156]: 😅 🤦‍♀️

**Pull request merged**